File: WebAPI/WebAPI.Models/Models/Models/Story.cs
using System;
using WebAPI.Models.Enums;

namespace WebAPI.Models.Models.Models
{
    public class Story
    {
        public Guid StoryId { get; set; }
        
        public Guid? UserId { get; set; }
        
        public Guid? SprintId { get; set; }
        
        public string Title { get; set; }
        
        public string Description { get; set; }
        
        public string Notes { get; set; }
        
        public ColumnType ColumnType { get; set; }
        
        public StoryPriority StoryPriority { get; set; }
        
        public UserPosition RequiredPosition { get; set; }
        
        public int Estimate { get; set; }
        
        public bool IsReady { get; set; }
        
        public bool IsBlocked { get; set; }
        
        public string BlockReason { get; set; }
        
        public DateTime CreationDate { get; set; }
        
        public uint RecordVersion { get; set; }

        public bool IsDeleted { get; set; }
    }
}

File: web/src/types/story.ts
import { Priority, StoryHistoryAction } from '../constants/storyConstants';
import { UserPosition } from '../constants/userConstants';

export interface IStory {
    storyId?: string;
    title: string;
    description: string;
    notes: string;
    recordVersion?: number;
    columnType: string;
    estimate: number;
    isReady: boolean;
    isBlocked: boolean;
    blockReason: string;
    creationDate?: Date;
    teamId?: string;
    userId?: string;
    sprintId: string;
    storyPriority: Priority;
    requiredPosition: UserPosition;
}

export interface IFullStory extends IStory {
    storyHistory: IStoryHistory[];
}

export interface ISelectedItem {
    key: string;
    value: string;
}

export interface IStoryDragAndDrop {
    columnTypeOrigin: string;
    columnTypeDestination: string;
    storyId: string;
}

export interface IStoryColumns {
    key: string;
    value: IStory[];
}

export interface IStoryHistory {
    storyHistoryId: string;
    storyHistoryAction: StoryHistoryAction;
    fieldName: string;
    previousValue: string;
    currentValue: string;
    userName: string;
    creationDate: Date;
}

export interface IStorySimpleModel {
    storyId: string;
    title: string;
    columnType: string;
    storyPriority: string;
    isReady: boolean;
    isBlocked: boolean;
    estimate: number;
    sprintId: string;
    recordVersion: number;
}
